vertx-consul-client/src/main/java/io/vertx/ext/consul/ConsulClient.java

```diff
@@ -609,6 +609,19 @@ static ConsulClient create(Vertx vertx, JsonObject config) {
   ConsulClient infoSession(String id, Handler<AsyncResult<Session>> resultHandler);
 
   /**
+   * Returns the requested session information
+   * This is blocking query unlike {@link ConsulClient#infoSession(String, Handler)}
```

Comment:
Let's review services blocking query:
```java
private void test() throws InterruptedException {

  // register new service (first one)
  runAsync(h -> writeClient.registerService(new ServiceOptions().setName("service1").setId("id1"), h));

  // get list of registered service
  // this is non-blocking query, just for know Consul state index
  ServiceList list1 = getAsync(h -> readClient.catalogServices(h));

  CountDownLatch latch = new CountDownLatch(1);

  // get list of registered service again.
  // this is blocking query, we will reseive answer from Consul when Consul state index will change
  // from id1=list1.getIndex() to id2>id1
  readClient.catalogServicesWithOptions(new BlockingQueryOptions().setIndex(list1.getIndex()), h -> {
    // collect names of registered services
    List<String> names = h.result().getList().stream().map(Service::getName).collect(Collectors.toList());
    // check that second service registered
    assertTrue(names.contains("service2"));
    // ok!
    latch.countDown();
  });

  // wait some seconds
  sleep(vertx, 2000);

  // and verify that blocking query is definitely blocking
  assertEquals(latch.getCount(), 1);

  // register new service again (second one)
  // when this service will be registered, Consul state index will be changed
  runAsync(h -> writeClient.registerService(new ServiceOptions().setName("service2").setId("id2"), h));

  // wait for blocking query ending
  awaitLatch(latch);

  // cleanup
  runAsync(h -> writeClient.deregisterService("id1", h));
  runAsync(h -> writeClient.deregisterService("id2", h));
}
```
Maybe something like this we should add to docs